nemo/collections/nlp/models/language_modeling/megatron_t5_adapter_model.py

```diff
@@ -159,56 +157,19 @@ def predict_step(self, batch: Any, batch_idx: int, dataloader_idx: int = 0) -> A
             encoder_input=None,
         )
 
-        processed_preds = []
-        processed_labels = []
-        processed_inputs = []
-
-        preds = predicted_token_ids.cpu().numpy().tolist()
-        enc_inputs = enc_input.cpu().numpy().tolist()
+        # Special ids to text function to handle stripping <eos> and special tokens with sentencepiece tokenizers.
+        preds_text = MegatronT5FinetuneModel.ids_to_text(predicted_token_ids, self.tokenizer)
+        input_text = MegatronT5FinetuneModel.ids_to_text(enc_input, self.tokenizer)
 
         if labels is not None:
-            labels = labels.cpu().numpy().tolist()
+            labels_text = MegatronT5FinetuneModel.ids_to_text(labels, self.tokenizer)
         else:
-            labels = [None] * len(preds)
```

Comment:
Love removing the redundant code. Can some of these same changes be done in `validation_step`? (probably in another PR though)

Reply:
Yeah, I'll leave this for another PR.